### ISSUE TITLE
[OLD] Legacy Spanish Translation Progress

### ISSUE THREAD

**Issue by tesseralis** [reactjs/es.react.dev#4]
# 🚨 Please Read this before contributing 🚨

## Maintainer List

- @alejandronanez 
- @carburo 
- @dmoralesm
- @Darking360

## For New Translators

To translate a page:

1. Check that no one else has claimed your page in the checklist and comments below. Do not create new issues claiming for new work please, let's keep everything here. If you already have an issue opened, please close it after your PR gets merged 🎉. Thanks!
2. Comment below with the name of the page you would like to translate. **Take only one page at a time**.
3. Submit a PR, assign @dmoralesm, @carburo, @Darking360 or @alejandronanez and add a link to it to your comment!

Please check the [style guide](https://github.com/reactjs/es.reactjs.org/blob/master/TRANSLATION.md#style-guide) to make sure translations are written in a consistent manner.

Also, check out [the Crowdin repository](https://crowdin.com/project/react/es) before starting the translation of a new article. **Click and open each article** to see how much work has been actually done. **"0%" in the index page does not mean the article has not been translated!**

# Work List

(🏃 Work in Progress; 👀 In Review)

## Core Pages

To do before releasing as an "official" translation. **Please translate these pages first.**

* [x] Home Page (@Zitle-Nancy) #123
* [x] Tutorial (@ivandevp) #83 

### Basics
  * [x] Hello World (@alejandronanez) #5
  * [x] Introducing JSX (@icarlossz) #8 
  * [x] Rendering Elements (@shmesa22) #9 
  * [x] Components and Props (@EzequielMonforte) #10 
  * [x] State and Lifecycle (@carburo) #62 
  * [x] Handling Events (@shmesa22) #77 
  * [x] Conditional Rendering (@ElRodrigote) #19 
  * [x] Lists and Keys (@EzequielMonforte) #22 
  * [x] Forms (@Darking360) #132 
  * [x] Lifting State Up (@sigfriedCub1990) #135 
  * [x] Composition vs Inheritance (@jesgarciamartinez) #150 
  * [x] Thinking in React (@sergiodxa) #27 

### API Reference
  * [x] React (@karolincastano) #139 
  * [x] React.Component (@semoal) #61 
  * [x] ReactDOM (@Darking360) #33 
  * [x] ReactDOMServer (@devrchancay) #110 
  * [x] DOM Elements (@LuisRevillaM) #59 
  * [x] SyntheticEvent (@michaelcavallaro03) #29
  * [x] Test Utilities (@d4vsanchez) #73
  * [x] Shallow Renderer (@tmjoseantonio) #130 
  * [x] Test Renderer (@elyalvarado) #37 
  * [x] JS Environment Requirements (@angelitaooo) #87
  * [x] Glossary (@dmoralesm) #1 https://github.com/reactjs/es.reactjs.org/pull/16

### Navigation

These are the navigation links that appear in the sidebar. Possibly wait until
the corresponding sections are translated to do these.

* [x] tutorial/nav.yml (@ivandevp) #83 
* [x] docs/nav.yml (@devrchancay) #155 

## Next Steps

These are the next most important translations, ordered by importance:

### Installation

* [x] Getting Started (@Darking360) #76 
* [x] Add React to a Website (@d4vsanchez) #148 
* [x] Create a New React App (@elyalvarado) #68 
* [x] CDN Links (@loconluis) #142 

### Hooks

Everyone is hooked on hooks! People are really excited for these and it would be great for them to be translated.

  * [x] Introducing Hooks (@KevinGomezDev) #156 
  * [x] Hooks at a Glance (@GiuMagnani) #70 
  * [x] Using the State Hook (@a13marquez) #60 
  * [x] Using the Effect Hook (@escorponox) #66
  * [x] Rules of Hooks (@icarlossz) #71
  * [x] Building Your Own Hooks (@EzequielMonforte) #34
  * [x] Hooks API Reference (@jvelezpo) #35
  * [x] Hooks FAQ (@caal-15) #151

### Warnings

These are the pages that you get when you click the links in the console (e.g. https://reactjs.org/warnings/dont-call-proptypes.html). People tend to search these, so **please don't translate the error message itself**.
 
* [x] dont-call-proptypes (@jvelezpo) https://github.com/reactjs/es.reactjs.org/pull/81
* [x] invalid-aria-prop (@jvelezpo) https://github.com/reactjs/es.reactjs.org/pull/81
* [x] invalid-hook-call-warning (@jpvg10) #141 
* [x] legacy-factories (@jpvg10) #138 
* [x] refs-must-have-owner (@jvelezpo) https://github.com/reactjs/es.reactjs.org/pull/81
* [x] special-props (@jvelezpo) #69 
* [x] unknown-prop (@jvelezpo) #69

### Advanced Guides

* [x] Accessibility (@ElRodrigote) #74 
* [x] Code-Splitting (@LuisRevillaM) #145
* [x] Context (@ricardoerl) #78
* [x] Error Boundaries (@jesgarciamartinez) #147 
* [x] Forwarding Refs (@elyalvarado) #72
* [x] Fragments (@RamirezAlex) #82 
* [x] Higher-Order Components (@elyalvarado) #125 
* [x] Integrating with Other Libraries (@jvelezpo) #175
* [x] JSX In Depth (@jvelezpo) #128 
* [x] Optimizing Performance (@RamirezAlex and @DanielGarc) #191 
* [x] Portals (@khriztianmoreno) #131 
* [x] React Without ES6 (@adielhercules) #118 
* [x] React Without JSX (@vickvasquez) #120 
* [x] Reconciliation (@devrchancay) #152 
* [x] Refs and the DOM (@Darking360) #86
* [x] Render Props (@jonathanpalma) #182
* [x] Static Type Checking (@Paolore) #136 
* [x] Strict Mode (@d4vsanchez) #126
* [x] Typechecking With PropTypes (@jnosornov) #113  
* [x] Uncontrolled Components (@Paolore) #117 
* [x] Web Components (@Paolore) #88 

### FAQ

* [x] APIs and AJAX (@kevinwolfcr) #2 
* [x] Babel, JSX, and Build Steps (@loconluis)
* [x] Passing Functions to Components (@florenciasilva) #124
* [x] Component State (@GiuMagnani) #75 
* [x] Styling and CSS (@GiuMagnani) #108 
* [x] File Structure (@GiuMagnani) #121 
* [x] Versioning Policy @alejandrogarciasalas  #154 
* [x] Virtual DOM and Internals @giossa94 #133 

### Contributing

Priority: Low

* [x] How to Contribute (@maudel) #153 
* [x] Codebase Overview - (@jnosornov) #186 
* [x] Implementation Notes (@devrchancay and @marcorichetta)
* [x] Design Principles (@elyalvarado) #146

### Components

Components in `src/components` that have some text in them.

* [x] CodeEditor (@icarlossz) #164 
* [x] CodeExample (@icarlossz) #164 
* [x] ErrorDecoder (@icarlossz) #164 
* [x] LayoutFooter (@Zitle-Nancy) #160 
* [x] LayoutHeader (@Zitle-Nancy) #160 
* [x] MarkdownPage (@icarlossz) #164 

### Community Resources

* [x] Support
* [x] Courses
* [x] Examples
* [x] Meetups
* [ ] Conferences
* [x] Articles
* [x] Podcasts
* [x] Videos
* [x] External Resources

### Tools

* [ ] Debugging
* [ ] Component Workbenches
* [ ] JSX Integrations
* [ ] Starter Kits
* [x] Routing
* [ ] Model Management
* [ ] Data Fetching
* [x] Testing
* [ ] UI Components
* [ ] Miscellaneous

### Blog

* [x] Is React Translated Yet? (@carburo)
* [ ] React v16.8
* [ ] React v16.7
* [ ] React 16 Roadmap
* [ ] React Conf Recap
* (and more...)

**Comment by shmesa22** [comment]
I would like to work on Introducing JSX

**Comment by alejandronanez** [comment]
Hey @shmesa22, I updated the description with your name on it! Thanks for helping us! 👍 

Introducing JSX is issue https://github.com/reactjs/es.reactjs.org/issues/7

Let us know if you have any questions!

**Comment by icarlossz** [comment]
Hey @shmesa22, I didn't see your comment and I started with that one. Sorry.

**Comment by EzequielMonforte** [comment]
Hi. I want to work on Components and Props

**Comment by shmesa22** [comment]
@icarlossz, no worries. I haven't started working on it, I can take another one

**Comment by shmesa22** [comment]
I would like to work on Rendering Elements then

**Comment by tesseralis** [comment]
@shmesa22 @EzequielMonforte are you able to edit the issue to add your name on it?

**Comment by shmesa22** [comment]
Hi @tesseralis, I was able to create the issue, but not to assign it to me

**Comment by shmesa22** [comment]
Hi everyone, any thoughts or any good translation for `callback`?

**Comment by sergiodxa** [comment]
I would like to work in `Thinking in React`

**Comment by sergiodxa** [comment]
@shmesa22 I'm not sure there is a good translation for that, the correct one would be `llamar de vuelta", maybe paraphrasing it it could fit in the sentence, however `callback` is a word commonly used in Spanish communities too so it may be an option avoid translating it.

**Comment by shmesa22** [comment]
Thank you @sergiodxa, that's what I was thinking. I left it like that.

Also, I would like to start working on handling events.

**Comment by tesseralis** [comment]
@shmesa22 Once it's done, we should have the glossary as a reference on how to translate common terms. @dmoralesm how is that progress going?

**Comment by shmesa22** [comment]
@tesseralis any way we can add or recommend words for the glossary?

**Comment by sergiodxa** [comment]
who is going to create the glossary? what's the criteria?

**Comment by tesseralis** [comment]
@shmesa22 The original English glossary or the Spanish version? The Spanish docs should just be translation of the English version.

@sergiodxa The "glossary" in question is the translation of the version in the English docs: https://reactjs.org/docs/glossary.html

Currently @dmoralesm is assigned to translate it but if it's not done soon we should hand over the work to someone else.

**Comment by dmoralesm** [comment]
> @shmesa22 Once it's done, we should have the glossary as a reference on how to translate common terms. @dmoralesm how is that progress going?

I'm about to finish

**Comment by sergiodxa** [comment]
I mean "a reference on how to translate common terms" that glossary.

**Comment by shmesa22** [comment]
Same here, I was thinking on some glossary on how to translate words like `callback` or `camelCase`

**Comment by sergiodxa** [comment]
mock too

**Comment by sergiodxa** [comment]
Or at least define what words not to translate because they are used as is in Spansh

**Comment by tesseralis** [comment]
We'll probably have a TRANSLATION folder with style-guide.md and other useful guides in there. Let me get that set up.

**Comment by sergiodxa** [comment]
Great!

I have another question, what about links to external sites? In Thinking in React there is a link to a Wikipedia page which has a Spanish translation, should I change the link to the Spanish version? or keep the English one?

**Comment by tesseralis** [comment]
@sergiodxa that seems like a good idea, but I'll leave it to more experienced translators. We can also put our decision in the style guide!

**Comment by sergiodxa** [comment]
Also what about text in block codes?
```
[
  {category: "Sporting Goods", price: "$49.99", stocked: true, name: "Football"},
  {category: "Sporting Goods", price: "$9.99", stocked: true, name: "Baseball"},
  {category: "Sporting Goods", price: "$29.99", stocked: false, name: "Basketball"},
  {category: "Electronics", price: "$99.99", stocked: true, name: "iPod Touch"},
  {category: "Electronics", price: "$399.99", stocked: false, name: "iPhone 5"},
  {category: "Electronics", price: "$199.99", stocked: true, name: "Nexus 7"}
];
```
That's also in Thinking in React, should I translate the categories and names? (when they are not direct product names) I read on the main issue they will be copied as if in the future they are modified on the original version, also since it's data it could be left as is as an example.

**Comment by tesseralis** [comment]
@sergiodxa maybe reference translations in other sites/languages to see what they do? Or see if you can find style guides from other technical projects we can use as a base for ours.

**Comment by gariasf** [comment]
> Also what about text in block codes?
> 
> ```
> [
>   {category: "Sporting Goods", price: "$49.99", stocked: true, name: "Football"},
>   {category: "Sporting Goods", price: "$9.99", stocked: true, name: "Baseball"},
>   {category: "Sporting Goods", price: "$29.99", stocked: false, name: "Basketball"},
>   {category: "Electronics", price: "$99.99", stocked: true, name: "iPod Touch"},
>   {category: "Electronics", price: "$399.99", stocked: false, name: "iPhone 5"},
>   {category: "Electronics", price: "$199.99", stocked: true, name: "Nexus 7"}
> ];
> ```
> That's also in Thinking in React, should I translate the categories and names? (when they are not direct product names) I read on the main issue they will be copied as if in the future they are modified on the original version, also since it's data it could be left as is as an example.

I would leave code as is with the exception of comments.

**Comment by dmoralesm** [comment]
> I would leave code as is with the exception of comments.

I agree with @gariasf

**Comment by tesseralis** [comment]
@sergiodxa @shmesa22 I added `TRANSLATION.md` as a source for translation information, including a style guide: https://github.com/reactjs/es.reactjs.org/blob/master/TRANSLATION.md. Feel free to edit and suggest style guide tips.

**Comment by dmoralesm** [comment]
> @sergiodxa @shmesa22 I added `TRANSLATION.md` as a source for translation information, including a style guide: https://github.com/reactjs/es.reactjs.org/blob/master/TRANSLATION.md. Feel free to edit and suggest style guide tips.

I would like to create a list of words that we must avoid to translate, such as *string* or *callback*. Is TRANSLATION.md a proper place to do it?

**Comment by tesseralis** [comment]
@dmoralesm Yup!

**Comment by EzequielMonforte** [comment]
I would like work on List and keys too #18

**Comment by manuelczs** [comment]
Hi, assign me:  Composition vs Inheritance

**Comment by icarlossz** [comment]
When I finish with introducing jsx I want to continue with lists and keys

**Comment by mbavio** [comment]
Hi, I would like to help translating `Composition vs Inheritance`.

**Comment by joelalejandro** [comment]
Hello! I'd love to help with `Forms`.

**Comment by sigfriedCub1990** [comment]
Hello there, I would like to help with `Lifting State Up`

**Comment by tmjoseantonio** [comment]
Hola!

I can help with Shallow Renderer 👌🏻

**Comment by karolincastano** [comment]
Hi, I’m an English to Spanish translator and I would like to help with the contributing page translation.

**Comment by semoal** [comment]
Hello! I'd love to help with `React`, or just the next going step on the priority list.

**Comment by elyalvarado** [comment]
Hello, I'd love to take `Test Renderer`. I opened issue #24

**Comment by KevinGomezDev** [comment]
Hey y'all, I would like to start with Hooks.

**Comment by LuisRevillaM** [comment]
I would love to help with the DOM elements section of the API Reference.

**Comment by GiuMagnani** [comment]
I'd love to help with the `Hooks at a Glance` translation.

**Comment by michaelcavallaro03** [comment]
I would love to help with,  SyntheticEvent translation!

**Comment by alejandronanez** [comment]
@michaelcavallaro03, @GiuMagnani, @LuisRevillaM, @KevinGomezDev all set.

**Comment by alejandronanez** [comment]
@icarlossz `List and Keys` has already been assigned to @EzequielMonforte.

> https://github.com/reactjs/es.reactjs.org/issues/4#issuecomment-459560954

If you're done with #8 (it says WIP still) you should grab another piece of documentation. 😄

**Comment by sergiodxa** [comment]
I created the PR for Thinking in React (#27) 

Now I want to work in Hooks FAQ.

**Comment by ivandevp** [comment]
I would like to help with the Tutorial section.

**Comment by sigfriedCub1990** [comment]
> > Hi, I’m an English to Spanish translator and I would like to help with the contributing page translation.
> 
> @karolincastano Every help is welcome. Right now the [style guide](https://github.com/reactjs/es.reactjs.org/blob/master/TRANSLATION.md#style-guide) is basically a stub. Feel free to propose changes to it.

@carburo, dónde está Guaso? Saludos de un hermano cubano

**Comment by alejandronanez** [comment]
Hey man, @sergiodxa, all set!

@ivandevp, alright, assigned it to you!

**Comment by a13marquez** [comment]
Hi everyone, i'd love to help in any section

**Comment by devrchancay** [comment]
I want to translate ReactDOMServer 🤓

**Comment by Zitle** [comment]
I want to translate Home :)

**Comment by Zitle-Nancy** [comment]
I want to translate Home :)

**Comment by d4vsanchez** [comment]
Hi! I’d like to help translating the “Test utilities” section.

**Comment by alejandronanez** [comment]
Thanks for your interest friends, welcome 🎉 

@d4vsanchez, @devrchancay and @Zitle-Nancy you're all set ✌️

**Comment by alejandronanez** [comment]
@a13marquez, please, choose one, we have plenty of work to do! 😄

**Comment by a13marquez** [comment]
@alejandronanez I can take Using the State Hook and Using the Effect Hook

**Comment by EzequielMonforte** [comment]
I want to work on #31

**Comment by jvelezpo** [comment]
i will work on `Hooks API Reference`

**Comment by jvelezpo** [comment]
special-props
unknown-prop

**Comment by alejandronanez** [comment]
Special props and Unknown Props are assigned to you @jvelezpo

**Comment by angelitaooo** [comment]
I want to work in JS environment requirements please!

**Comment by alejandronanez** [comment]
You got it @angelitaooo

**Comment by escorponox** [comment]
Let me take the `Using the Effect hook` if it is still available

**Comment by Darking360** [comment]
Greetings :rocket: 

I just created a PR for ReactDOM here: [ReactDOM translation](https://github.com/reactjs/es.reactjs.org/pull/33)

I'd love to keep helping, maybe I can help with the `Getting Started` :pray: let me know and I'll start it right away :+1: @dmoralesm

**Comment by anyk1709** [comment]
Hey! I'd like to help with Add React to a Website

**Comment by alejandronanez** [comment]
Hello 👋 @Darking360 and @anyk1709, thanks for your interest!

I already assigned you the issues you mentioned! 👍

**Comment by elyalvarado** [comment]
Having already finished with #24, and it already merged, I'll like to work on `Create a New React App`

**Comment by ElRodrigote** [comment]
Hi, I'm finishing #17 and would like to take `Accessibility` as I've worked with that in my current job :)

**Comment by ricardoerl** [comment]
I would like to help with `Context`

**Comment by jonathanpalma** [comment]
I'd like to help with `Render Props`

**Comment by carburo** [comment]
Thank you for your interest.. :tada: Context goest to @ricardoerl & Render Props to @jonathanpalma.

**Comment by RamirezAlex** [comment]
I would like to work on `Fragments` :-)

**Comment by LuisRevillaM** [comment]
I would like to translate Code-Splitting on Advanced Guides.

**Comment by carburo** [comment]
Welcome @RamirezAlex! I alredy assign you Fragments. @LuisRevillaM, @elyalvarado and @ElRodrigote I already assign you the issues you mentioned.

**Comment by ElRodrigote** [comment]
> Welcome, @RamirezAlex! I already assign you Fragments. @LuisRevillaM, @elyalvarado, and @ElRodrigote I already assign you the issues you mentioned. Remember not to open new issues for translations.

Thanks! Didn't know, sorry, won't happen again :)

**Comment by elyalvarado** [comment]
Done with `Forwarding Refs`. PR is #72 @dmoralesm, @carburo and @alejandronanez.

I'll like to work on `Higher-Order Components`

**Comment by alepaz** [comment]
Hi! Let me help with `Integrating with Other Libraries`

**Comment by alejandronanez** [comment]
Ok, friends, we have:

- @GiuMagnani, I assigned to you `Component State`. I didn't assign `Styling and CSS` to you just yet, we're trying to have 1 PR per person at the time, also someone might be interested in `Styling and CSS` while you work in `Component State`. Once you're done with `Component State` let us know and we can assign `Styling and CSS` (or any other) to you! 👍 
- @elyalvarado you got `Higher-Order Components`
- @alepaz you got `Integrating with Other Libraries`

**Comment by GiuMagnani** [comment]
Thanks @alejandronanez, I just sent the PR for the `Component State` FAQ: [#75](https://github.com/reactjs/es.reactjs.org/pull/75). Let me know when I can take the `Styling and CSS` or another that is available. Thanks again!

**Comment by jesgarciamartinez** [comment]
Hi! I'd like to work on _Error Boundaries_

**Comment by jvelezpo** [comment]
Hello, i will work on 
* refs-must-have-owner (this is a short file so i will take on other file)
* invalid-aria-prop (this one is also a short file)
* dont-call-proptypes

@alejandronanez  One question i dont see `legacy-properties` but i see `legacy-factories` file, if this is a typo on the description of the issue, i can take on `legacy-factories` as well

**Comment by RamirezAlex** [comment]
I open a PR #82 for `Fragments` to close issue #80 
I would like to continue with `Optimizing Performance`

**Comment by alejandronanez** [comment]
@jvelezpo thanks for your enthusiasm, but, please, take one issue at the time. It doesn't matter if they are small ones. Once you're done with one issue, we can assign you another.

**Comment by florenciasilva** [comment]
Hi! I would like to work on **FAQ: Passing Functions to Components.**
https://github.com/reactjs/es.reactjs.org/blob/master/content/docs/faq-functions.md

**Comment by Darking360** [comment]
Hey! I just finished the PR for the `Getting Started` page here: [Getting Started PR](https://github.com/reactjs/es.reactjs.org/pull/76) 🚀 

I'd love to continue with: `Refs and the DOM` if that's ok. Please let me know, and I'll probably start it tomorrow 💪 

**CC:** @tesseralis @dmoralesm @carburo @alejandronanez

**Comment by alejandronanez** [comment]
Hello!

@florenciasilva you got `FAQ: Passing Functions to Components`
@RamirezAlex I assigned `Optimizing Performance`
@jvelezpo you got  `refs-must-have-owners`, `invalid-aria-prop`, `dont-call-proptypes`. Once you're done with those 3, you can claim `JSX in Depth` if it is still available 👍 
@Darking360 `Refs and the DOM` is yours

Thank you friends!!!

**Comment by jvelezpo** [comment]
@alejandronanez done with `refs-must-have-owners`, `invalid-aria-prop`, `dont-call-proptypes`
on PR https://payfully.atlassian.net/browse/PF-711
i would like `JSX in Depth` if it is still available

**Comment by RamirezAlex** [comment]
Hi @alejandronanez @carburo @tesseralis and others: I think it is very important that we establish which form of second-person treatment we are going to use for consistency. Meaning, I have seen some contributors using `tu` others using `vos` and others including me using `usted`. :D

In my case I started using `usted` because I consider this formal documentation, thus I use formal treatment and it is more neutral. However, I feel Ok using other as soon as it is consistent through the entire translation. What do you think? Peace 👍

**Comment by GiuMagnani** [comment]
I vote for `tú`, I don't think a library's documentation should be that formal.

**Comment by elyalvarado** [comment]
I agree that the tone in English is kind of informal, specially because it isn't using the third person. That's what I think that the more familiar/informal `tú` matches better the original intent.

**Comment by Paolore** [comment]
Hello, I would like Web Component.

**Comment by alejandronanez** [comment]
@RamirezAlex, Echoing what @GiuMagnani @elyalvarado and @sergiodxa said, let's go with `tú`!

**Comment by alejandronanez** [comment]
👋 @Paolore you got `Web Components` and @jvelezpo you got `JSX in Depth`!!!

Thanks for all your help here!

**Comment by tesseralis** [comment]
@alejandronanez if you have a style guide you can put "use _tú_" in it!

**Comment by Paolore** [comment]
Web Component translation #88
I could continue with  Uncontrolled Components

**Comment by carburo** [comment]
So, Strict mode for @d4vsanchez and Uncontrolled Components for @Paolore. 
Good job everybody!

**Comment by GiuMagnani** [comment]
Done with `Styling and CSS` @dmoralesm, @carburo and @alejandronanez, PR is #108. Can I take `File Structure` from FAQ now?

**Comment by escorponox** [comment]
Done with `Using the Effect Hook` @dmoralesm, @carburo and @alejandronanez. #66

**Comment by jnosornov** [comment]
I would like to work on Typechecking With PropTypes

**Comment by adielhercules** [comment]
I would like to work on React Without ES6

**Comment by jnosornov** [comment]
I would like to work on Typechecking With PropTypes #111

**Comment by adielhercules** [comment]
I would like to also work on React Without JSX #112

**Comment by carburo** [comment]
Ok, 
- @GiuMagnani I already assign you File Structure.
- @adielhercules I gave you React without ES6, once you are done with it, feel free to claim another one.
- @jnosornov I assign you Typechecking with PropTypes.

Do not create new issues claiming for new work please, let's keep everything here.
Thank you all!

**Comment by tesseralis** [comment]
Hi all!

I just merged in changes from the hook release from master, and there are a few files that need to be updated. You can see the PR here: https://github.com/reactjs/es.reactjs.org/pull/115

The files that need to be updated (and the original translators):

 content/docs/addons-test-utils.md (@d4vsanchez)
  content/docs/hooks-custom.md (@EzequielMonforte)
  content/docs/hooks-reference.md (@jvelezpo)

**Comment by alejandronanez** [comment]
Heads up @d4vsanchez, @EzequielMonforte and @jvelezpo we just merged #115!

Let us know if you need any help!

**Comment by devrchancay** [comment]
I want to translate reconciliation 🤓😬

**Comment by Paolore** [comment]
I want  React Without JSX
:)

**Comment by carburo** [comment]
Hello @joelalejandro, @sigfriedCub1990, @karolincastano, @tmjoseantonio. There has been almost a week since you requested one of the articles. As you haven't yet opened a PR we wanted to know what's the current status of your work on those articles. Also, if for any reason you cannot continue, please let us know in order to pass it to some one else.

**Comment by carburo** [comment]
@devrchancay, Reconciliation is yours now.
@Paolore I didn't assign you React Without JSX because you have already a translation for Uncontrolled Components. Let us know when you finish and we will be happy to give you another one.

**Comment by GiuMagnani** [comment]
Done with `File Structure` @dmoralesm, @carburo and @alejandronanez, PR is #121. Thanks!

**Comment by Paolore** [comment]
Hi @carburo, yes, then I want Static-Type-Checking. thanks!

**Comment by adielhercules** [comment]
I've just created a PR for React Without ES6 cc @carburo

**Comment by khriztianmoreno** [comment]
I'd love to help with the ` Portals` translation. @alejandronanez

**Comment by carburo** [comment]
Ok @khriztianmoreno. You have it!

**Comment by SebPez** [comment]
I'd like to help with Anything from FAQ's. Ideally ` Babel, JSX, and Build Steps` + something else.

**Comment by jnosornov** [comment]
made some changes suggested by @RamirezAlex to PR #113. Can anyone else take a look to see if further mods are need it?.

**Comment by alejandronanez** [comment]
You got it @SebPez!

@jnosornov, thanks! For the next occasion, you can tag us in the PR and Github will notify us :)

**Comment by tmjoseantonio** [comment]
@carburo,

@karolincastano and I are going to start pushing updates during this weekend, if you consider our assignments can be passed to another people please go ahead as we have been really busy these last days. Apologies for that.

**Comment by carburo** [comment]
@tmjoseantonio & @karolincastano. No problem! We just wanted to check because those pages are needed to have an MVP. I'll leave it assigned to you then.

**Comment by carburo** [comment]
Hello @Zitle-Nancy, @mbavio. There has been almost a week since you requested one of the core pages. As you haven't yet opened a PR we wanted to know what's the current status of your work on those articles. Also, if for any reason you cannot continue, please let us know in order to pass it to someone else.

**Comment by carburo** [comment]
I am opening up the slot of Forms that was assigned to @joelalejandro. Please, claim it if you are going to start working on it, as it is one of the core pages needed to have a minimum viable translation.

**Comment by Darking360** [comment]
@carburo I could start working on the `Forms` since I'm finishing the `Refs and the DOM` translation, I could start it during the day, and have it finished for tomorrow if that's ok. Let me know :rocket: 

**CC:** @alejandronanez @dmoralesm

**Comment by andfelzapata** [comment]
I would like to work on the Contributing section.

**Comment by jnosornov** [comment]
@carburo If there is any trouble translating some of the core pages, just let me know. I guess `Forms` would be assigned to @Darking360 since he claimed it already.

**Comment by elyalvarado** [comment]
Finished translation of `Higher-Order Components` on PR #125. I'll appreciate any comments/corrections as it was not only long, but also the concept is hard to understand for many people and I want to make sure the translation is ok.

I'll shoot myself in the foot again, and would like to take `Virtual DOM and Internals` @carburo @alejandronanez @dmoralesm

**Comment by mbavio** [comment]
@carburo Working on it, will have a WIP PR later today.

**Comment by giossa94** [comment]
I'd like to translate **Virtual DOM and Internals**, which belongs to the FAQ section.

**Comment by Zitle-Nancy** [comment]
@carburo Hello, sorry for the delay, I just send my PR (#123), in this morning and  I am waiting for your feedback.
Cheers!

**Comment by alejandrogarciasalas** [comment]
I'd like to help with `Versioning Policy`! 
@dmoralesm, @carburo  @alejandronanez

**Comment by carburo** [comment]
Ok, @Darking360 you got Forms

**Comment by loconluis** [comment]
I would like to work on the **CDN Links** and also on **How to Contribute**

**Comment by jpvg10** [comment]
Hello, I would like to do `Legacy factories` in the warnings section @dmoralesm, @carburo @alejandronanez

**Comment by tesseralis** [comment]
You got it @jpvg10 !

**Comment by elyalvarado** [comment]
@alejandronanez @carburo @dmoralesm PR #125 is ready to be merged, I'll like to work on `Design Principles`

**Comment by MiguelMadero** [comment]
Can I do Design Principles ?

**Comment by maudel** [comment]
@dmoralesm, @carburo @alejandronanez I really want to help I can work on any page you need help .
These ones for example doesn't have a person in charge. 

* How to Contribute
* [ ]  Codebase Overview
* [ ]  Implementation Notes
* [ ]  Design Principles

**Comment by Paolore** [comment]
Hi @carburo, is ready Static-type-checking to review. PR 136

**Comment by carburo** [comment]
Ok, @loconluis you got "CDN Links". After you finish that you can claim another one.
@elyalvarado You got Design Principles
@maudel "How to Contribute"
@MiguelMadero That one was claimed already by @elyalvarado.  Would you like to take "Codebase Overview" or "Implementation Notes"?

**Comment by jpvg10** [comment]
`Legacy factories` ready for review! 

Also I noticed that in the warnings folder there is a file that is not in the checklist on this issue: `invalid-hook-call-warning`

**Comment by carburo** [comment]
Yes, @jpvg10, you are right. Thank you! This was added for React 16.8. Do you want to take it?

**Comment by jpvg10** [comment]
Ok @carburo I'll take it

**Comment by jpvg10** [comment]
`Invalid hook call warning` is ready for review!

**Comment by MiguelMadero** [comment]
@carburo I can do "Codebase Overview" or "Implementation Notes" in that order.

**Comment by escorponox** [comment]
I created a new PR for a fix on `Effect Hooks` #144 
@carburo

**Comment by elpopisencio** [comment]
I'd like to help but apparently I'm too late. Let me know if there's anything I can do!

**Comment by capradenas** [comment]
> I'd like to help but apparently I'm too late. Let me know if there's anything I can do!

:( me too

**Comment by carburo** [comment]
@MiguelMadero you got "CodebaseOverview"
@elpopisencio is not too late, you got "Implementation Notes"
@capradenas It's never too late. These pages have to be continually improved and updated from upstream. With that said, the components section is still available. There is little translatable text in there, but it is something that needs to be done as well. Everyone can help to fix issues they spot in existing articles or reviewing open PRs. Also, we are pinging people in charge of some articles that seem to be abandoned so we can make them available for anyone interested. Tell me if you are interested in translating some of the components of if you have anything else in mind.

**Comment by carburo** [comment]
@mbavio Sorry to bother you again, it has been 11 days since you took the article "Composition vs Inheritance" and is the only one of the core articles that doesn't have a PR open yet. Could you open at least a "Work in Progress" PR? If you cannot finish there is no problem, we can open the spot for someone else to continue the work.
Also, @LuisRevillaM,  @anyk1709, @KevinGomezDev, @sergiodxa, @alepaz  we wanted to know your current progress or if you are not able to continue.

**Comment by caal-15** [comment]
Hey if there's something for me to do, I can take anything you have left :D

**Comment by d4vsanchez** [comment]
Me too. If I can help with something else on the translation matter, just hit me up.

**Comment by sergiodxa** [comment]
@carburo I don't think I'm going to be able to work on Hooks FAQ, too busy right now.

If @caal-15, @d4vsanchez or someone else wants to take it go ahead 👍

**Comment by caal-15** [comment]
I can take it and start working on it tonight if that sounds cool? :)

**Comment by elyalvarado** [comment]
The complete draft for the translation of `Design Principles` is ready on PR #146 for review. I'll highly appreciate anybody's suggestions as it was a very long article with a lot of complex concepts without a direct translation, so I had to take some liberties.

**Comment by mbavio** [comment]
@carburo Can't get a breath from work in order to work on `Composition vs Inheritance`. Feel free to assign it to somebody else.

**Comment by jesgarciamartinez** [comment]
The PR for the translation of `Error Boundaries` is up - PR [#147](https://github.com/reactjs/es.reactjs.org/pull/147).

I could do `Composition vs Inheritance` next.

**Comment by icarlossz** [comment]
I can take the remaining components.

**Comment by carburo** [comment]
@caal-15 You got Hooks FAQ
@jesgarciamartinez I assign you Composition vs Inheritance, note this is a core article, and we are trying to push these asap. So, if you are don't think you will able to have a PR soon, please let us know.
@icarlossz you got the remaining components
@d4vsanchez We can assign you Add React to a website that is now open. Let us know.

**Comment by d4vsanchez** [comment]
@carburo Yes, I'll take "Add React to a website". Thanks!

**Comment by jvelezpo** [comment]
Hello @carburo  i am seeing this `Hooks API Reference (@jvelezpo) #35 (Needs updating)`

can you explain me what is needs updating, i am assuming there were some changes in the original repo  that and they need translation, am I right?

**Comment by ivandevp** [comment]
I've just finished translating the tutorial (#84). Is there anything else that I can help? I see the blog and community section, but let me know if there's anything with a high priority.

**Comment by carburo** [comment]
@jvelezpo Yeah, sorry. They are now fully updated. Some changes were already made. The only missing part of the articles was to add the heading IDs. Thank you for pointing that out.

**Comment by d4vsanchez** [comment]
I've finished translating `Add React to a Website`.

Waiting for the review 😄 @alejandronanez @dmoralesm @carburo

**Comment by jesgarciamartinez** [comment]
The PR for the translation of `Composition vs Inheritance` is up - PR [#150](https://github.com/reactjs/es.reactjs.org/pull/150).

**Comment by LuisRevillaM** [comment]
PR #145 `Code-Splitting` is ready to be reviewed. Thank you!

**Comment by caal-15** [comment]
Added WIP PR for Hook FAQ translation: #151, will update when ready.

**Comment by devrchancay** [comment]
The PR for the translation of `Reconciliation` #152

**Comment by jvelezpo** [comment]
Hello @carburo,

i see some tasks like `* Introducing Hooks`,  `Hooks FAQ` or `Integrating with Other Libraries` (and others) assigned many days ago and still there is no progress on them.

I the tasks are going to be reassigned let me know, i will be happy to work on them 👍

**Comment by KevinGomezDev** [comment]
Hi @carburo I will send the PR for `Introducing Hooks`  today or tomorrow afternoon

**Comment by elyalvarado** [comment]
As I didn't receive any comments I'll mark #146 as done and ready for merging. @carburo @alejandronanez

**Comment by elyalvarado** [comment]
Hi @everybody, I just created an #es channel on the [React Translations Slack](https://join.slack.com/t/react-translations/shared_invite/enQtNTQ4MDQxMDQwMTk1LWRjMWI0ZjVlYzUyYmMzN2Q5MmJiNzA1MjdlNjViM2Y0MDQ2ZTRhZmM5NTliYmU1MzRmMTFhNzhkMmU3OTg3MjY) created by @tesseralis. This is a great place to have a more interactive discussion and to get real-time help. See you there.

**Comment by tesseralis** [comment]
Actually @telmogoncalves made the slack, I just made it official :)

**Comment by telmogoncalves** [comment]
> Actually @telmogoncalves made the slack, I just made it official :)

The only reason I'm here it's you. If it wasn't for your Tweet I wouldn't be here, so, thank you @tesseralis

**Comment by ivandevp** [comment]
Hey @alejandronanez , the `tutorial/nav.yml` was translated in #83, too. So, I think you can check it as done in the navigation section of this issue's body.

**Comment by carburo** [comment]
I am opening "Integrating with Other Libraries" as we've got no answer from the @alepaz. Would you like to take it @jvelezpo?
Also, we want to know from the current status of the translations carried by @SebPez, @RamirezAlex and @alejandrogarciasalas. If you cannot carry on with them we will assign them to other contributors.

**Comment by jvelezpo** [comment]
yes i can take "Integrating with Other Libraries" @carburo 👍

**Comment by ivandevp** [comment]
Hey @carburo, let me know if I can help with any other part of translation, please.

**Comment by Darking360** [comment]
Greetings! 🚀 

Just finished the translation for [Forms](https://github.com/reactjs/es.reactjs.org/pull/132)

@carburo If there's another part that I can help to translate, let me know 🙏

**Comment by maudel** [comment]
Just Finish the translations of [how to contribute](https://github.com/reactjs/es.reactjs.org/pull/153)

@carburo 

the same for my if I can help translating another page please let me know

**Comment by alejandrogarciasalas** [comment]
@carburo I am done with the translation in [crowdin](https://crowdin.com/translate/react/2944/en-es). I'll make the PR in a bit. Sorry about the delay 🙇‍♂️

**Comment by alejandrogarciasalas** [comment]
Update: PR translating `faq-versioning` #154. Thanks @carburo !

**Comment by loconluis** [comment]
I could also help if something was needed.

**Comment by devrchancay** [comment]
I am translating `docs/nav.yml`

**Comment by KevinGomezDev** [comment]
@carburo @alejandronanez  Introducing Hooks PR is ready for review #156
Sorry for the Delay, a lot of work these days.

**Comment by RamirezAlex** [comment]
@carburo I will submit my PR soon, it is a long one.

**Comment by tesseralis** [comment]
You all are amazing! I made a final list of things to do to prepare the site for release! https://github.com/reactjs/es.reactjs.org/issues/158

**Comment by Zitle-Nancy** [comment]
Hi @carburo , I finished the translation of <LayoutFooter /> and <LayoutHeader />  (#160)
Let me know if you have any comments.

**Comment by tesseralis** [comment]
WE DID IT!!!
<img width="364" alt="screen shot 2019-02-16 at 5 40 43 pm" src="https://user-images.githubusercontent.com/1278991/52907235-47634900-3212-11e9-8f67-9cb736015f7b.png">

**Comment by carburo** [comment]
I am opening "Babel, JSX, and Build Steps" as it seems to be abandoned. So anyone can claim it now.

**Comment by loconluis** [comment]
@carburo I can take it.

**Comment by carburo** [comment]
Ok, @loconluis. You got it. Thank you!

**Comment by ricardoerl** [comment]
Hi @alejandronanez, I finished the translation for (#78)
Sorry for the delay, Thanks.

**Comment by carburo** [comment]
We want to know from the current status of the translations carried by @MiguelMadero and @elpopisencio. Let us know, If you cannot carry on with them we can assign them to other contributors.

**Comment by elpopisencio** [comment]
Hi @carburo. I can't work on this right now. I'm really sorry. Please assign it to someone else.

**Comment by devrchancay** [comment]
@carburo I can take the issue of *Implementation Notes* that was from @elpopisencio

**Comment by Darking360** [comment]
@devrchancay It's yours! 🚀

**Comment by carburo** [comment]
We want to know from the current status of the translation carried by @jonathanpalma. If you cannot carry on we can assign it to someone else.

**Comment by MiguelMadero** [comment]
Hi @carburo sorry for the delay. Life is getting in the way of setting time aside for this. I likely won't get to it until next weekend. Please re-assign if needed and I can find other opportunities to contribute later.

**Comment by alejandronanez** [comment]
Thanks for letting us know @MiguelMadero!

**Comment by jnosornov** [comment]
Would like to work on  Codebase Overview - up for grabs @alejandronanez @carburo

**Comment by carburo** [comment]
Ok @jnosornov, you got it. Thank you!

**Comment by jonathanpalma** [comment]
Hey @carburo, sorry for the delay but here is the translation for Render Props #182

**Comment by jnosornov** [comment]
Hey guys, DOUBT... lets say we have this url (/docs/reconciliation.html), that points to the English docs, what would be right one for the Spanish docs.

**Comment by alejandronanez** [comment]
Hey @jnosornov, it should be the same url. We’re leaving permalinks untouched. 👍🏼

Thanks for asking,

**Comment by jnosornov** [comment]
Hey guys, @carburo @alejandronanez  here is the PR for **Codebase Overview** #186. Check it out and let me know the changes, ill do them ASAP.

**Comment by jnosornov** [comment]
Hey guys, suggested changes to #186 have been made. Thanks to @icarlossz, @Darking360  for the review, and also @alejandronanez for requesting a review.

**Comment by marcorichetta** [comment]
Hi!
First of all, congrats everyone on this great job!

I'm looking on the [spanish page](https://es.reactjs.org/) and the navbar item 'Comunidad' it's cut by the search box.

This seems to be because the navbar has the same structure on the 2 pages (**English and Spanish**), but in spanish the words are longer.

![spanish nav](https://user-images.githubusercontent.com/19599150/53670301-0844e700-3c59-11e9-89ed-8b77394c2cd7.png)

![english nav](https://user-images.githubusercontent.com/19599150/53670304-0c710480-3c59-11e9-884a-b35509621c98.png)

**Comment by Darking360** [comment]
Greetings @marcorichetta And yes, I'm seeing that too 🤔 It's because `Docs` being translated to `Documentación`, let's see what we can do.

**CC:** @alejandronanez @carburo @tesseralis

**Comment by alejandronanez** [comment]
hmmmm interesting - I might be able to take a look at it on Sunday I think cc @darking360 @carburo

**Comment by tesseralis** [comment]
You can scroll it to see the whole thing but yeah it's not ideal. Some ideas:

* Change "Documentación" to "Docs"
* Make the padding in between the words smaller
* Shift the nav bar to the left like so:

<img width="1276" alt="screen shot 2019-03-01 at 3 47 10 pm" src="https://user-images.githubusercontent.com/1278991/53672878-54802f00-3c39-11e9-9a1b-62e04a3f41e3.png">

The last one is probably the best for now.

**Comment by alejandronanez** [comment]
Hey @carburo @Darking360 @tesseralis, I don't have enough bandwidth this week to address the issue. Please, feel free to jump in!

**Comment by ivandevp** [comment]
I can help with that fix if you want, let me know if you need any help, please.

**Comment by DanielGarc** [comment]
Hello I noticed  **🏃 Implementation Notes** and **🏃 Optimizing Performance** are not done yet, is it ok for me to work on either of those?

**Comment by RamirezAlex** [comment]
Hey @DanielGarc Please take a look at this one and go ahead to continue with it. #191 :)

**Comment by Darking360** [comment]
Greetings people, sorry for the delay. So in order:

1.- @ivandevp Please if you could start working on that issue and submit a PR would be awesome! 💪 
2.- @DanielGarc Start working on `Optimizing Performance` as @RamirezAlex asked you, so you can finish it, I'll ping the one that has `Implementation Notes` to see if you could address that issue as well, but I'll let you know later.
3.- @devrchancay How are you? How's it going with `Implementation Notes`, could you still finish it? Let us know 👀 

That would be all for now, after finish those items then we'd have to translate the blog and see what we're gonna do with `Community` but let's talk that afterward, let me know any questions 🚀

**Comment by devrchancay** [comment]
@Darking360 Hi, it seems that this topic is unlucky, I'm on vacation and I have little internet access, please if you can assign it to someone else.

> I made 25% of the document will not be very helpful so do not publish it.

**Comment by Darking360** [comment]
@DanielGarc Greetings, well sir you got yourself another ticket, please start working on `Implementation Notes` too and create a PR when you have it 🚀 

@devrchancay Perfect, enjoy your vacations man 🌴 🍸 🥝

**Comment by marcorichetta** [comment]
Since @devrchancay is on vacation, I could help on `Implementation Notes` or `Optimizing Performance` too.
How do you organize between 2 or more people working on the same article?

**Comment by DanielGarc** [comment]
@marcorichetta I don't mind if you work on `Implementation Notes` as I haven't started working on it. 

**CC**: @Darking360

**Comment by Darking360** [comment]
Perfect guys. @marcorichetta you got `Implementation Notes` 🚀

**Comment by marcorichetta** [comment]
Hey @Darking360 @carburo @alejandronanez @dmoralesm, #196 is ready for review!

Thanks!

**Comment by tesseralis** [comment]
I updated the issues list so people can pick at the Community and Blog sections -- I also have an [issue here](https://github.com/reactjs/reactjs.org-translation/issues/84) to discuss what's the best way to approach these translations!

(side note it would be super cool if the blog post I made about translations got translated!)

**Comment by carburo** [comment]
I'll start working on it. I already had it on my TODO list.

**Comment by alejandronanez** [comment]
@carburo are you opening a new issue like #4 - I guess, right?

Thanks btw!

**Comment by carburo** [comment]
@alejandronanez, yeah, I think maybe this has become some sort of "meta-issue" and it's too big. What do you think about opening a new issue only for the blog?

**Comment by tesseralis** [comment]
I’m fine with creating new issues for the blog/community!

**Comment by RamirezAlex** [comment]
I am back on the game! Sorry I was off last weeks. Can I work on `support`? I can also help reviewing :)

**Comment by alejandronanez** [comment]
Hey @alexramirez, welcome back!

You can claim your issue in #201 :)

**Comment by jvelezpo** [comment]
Hello, i am back too 😎
can i work on `Examples`?

**Comment by tesseralis** [comment]
¡Felicidades!

<img width="340" alt="Screen Shot 2019-03-17 at 1 25 09 PM" src="https://user-images.githubusercontent.com/1278991/54497514-cbaff700-48b8-11e9-8f2b-eacdda621746.png">

**Comment by ramadis** [comment]
Hey guys, I've just pushed a PR (#281) with a few small fixes in the translation of the `typechecking-with-proptypes` page.

**Comment by matiasfh** [comment]
I would like to work on blog posts

**Comment by Darking360** [comment]
Hey @matiasfh we have an issue for the blog posts here: https://github.com/reactjs/es.reactjs.org/issues/201 🚀

**Comment by pablomdd** [comment]
Hey guys, just added some minor typo fixes in `rendering-elements` and `components-and-props` in PR #371

**Comment by apenab** [comment]
I would like work in 
Blog>React v16.8

**Comment by carburo** [comment]
@apenb The open issue for blog posts is #201. I assigned it to you there.

**Comment by FranciscoImanolSuarez** [comment]
I translated the document 2020-10-20-react-v17.md, if nobody else did I would like to open PR

**Comment by carburo** [comment]
@FranciscoImanolSuarez Sure! We will be glad to receive your PR!
I assign it to you on issue #201, so no one else starts working on it.

**Comment by FranciscoImanolSuarez** [comment]
@carburo I already sent the PR

**Comment by EvanErickson** [comment]
What issues are still open? The one's that haven't been checked off? I'm willing to translate a few pages.

**Comment by pmirandaarias** [comment]
I would like to edit https://github.com/reactjs/es.reactjs.org/blob/main/content/community/team.md it has spanish a typo

**Comment by gaearon** [comment]
Thanks to everyone who contributed! Issue for the new site: https://github.com/reactjs/es.react.dev/issues/500
